spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -151,4 +161,25 @@ TableCatalog tableCatalog() {
       return tableCatalog;
     }
   }
+
+  /**
+   * Closes the procedure's executor service if a new one was created.
+   */
+  protected void closeService() {
+    if (executorService != null) {
+      executorService.shutdown();
```

Comment:
Sure we could set it to null, again these are pretty private methods so I'm not to worried about usages. 

I'm not sure what you are asking for with the synchronize here, do you want it to be
shutdown?
awaitTermination?

I think in this usage that the pool had better be out of work by the time we get to this point since we have already returned results but we can do that

Reply:
I can add a 
 Preconditions.checkArgument(executorService.isTerminated(),
          "Procedure error, close service called before service was finished");
          
          Which will cause this to break if it is used incorrectly by iceberg devs

Reply:
The mention of `synchronize` can be ignored. I was just covering all my bases and was mostly concerned about `closeService` getting called multiple times. But I can see how that would be unlikely to happen.